<launch>
    <rosparam file="$(find ros_tcp_endpoint)/config/params.yaml" command="load"/>
    <param name="TCP_NODE_NAME" type="string" value="random" />
    <param name="PREFIX" value="/server_endpoint_" />
    
    <node name="server_endpoint_lidar0" pkg="ros_tcp_endpoint" type="lidar_server_endpoint.py" args="--wait" output="screen" respawn="true" />
    <node name="lidar0_converter" pkg="pcl2_converter" type="pcl2_converter_node" output="screen">
	<remap from="/pcl" to="/lidar/scan0"/>
        <remap from="/pcl2" to="/lidar/scan0/pcl2"/>
    </node>


    <!-- <node name="server_endpoint_lidar1" pkg="ros_tcp_endpoint" type="lidar_server_endpoint.py"  output="screen" respawn="true" />

    <node name="server_endpoint_global_pcl0" pkg="ros_tcp_endpoint" type="lidar_server_endpoint.py"  output="screen" respawn="true" /> -->


    <node name="rviz" pkg="rviz" type="rviz" output="log" args="-d $(find ros_tcp_endpoint)/config/pcl.rviz" />
</launch>
